framework/decode/dx12_dump_resources.cpp

```diff
@@ -2621,9 +2630,11 @@ void DefaultDx12DumpResourcesDelegate::TestWriteFloatResource(const std::string&
             data += "\n";
         }
 
-        const char* suffix        = Dx12DumpResourcePosToString(resource_data->dump_position);
-        std::string file_name_sub = file_name + "_sub_" + std::to_string(sub_index) + "_" + suffix + ".txt";
-        std::string file_path     = gfxrecon::util::filepath::Join(json_options_.root_dir, file_name_sub);
+        std::string suffix = Dx12DumpResourcePosToString(resource_data->dump_position);
+        std::string file_name_sub =
+            file_name + "_sub_" + std::to_string(sub_index) + (suffix == "" ? ".txt" : ("_" + suffix + ".txt"));
+        file_name_sub += (suffix == "" ? ".txt" : ("_" + suffix + ".txt"));
```

Comment:
Add ‵(suffix == "" ? ".txt" : ("_" + suffix + ".txt"))` twice.

Reply:
Good find. Fixed in subsequent commit. I don't have good test for this case, can you supply one?

Will squash this commit before merging to dev.